### ISSUE TITLE
Prettier removing semi-colons

### ISSUE THREAD

**Issue by Dropboy** [jsfiddle/jsfiddle-users#1896]
Hey not sure if this is intended but it looks like semi-colons are being removed in JS tab when using prettier to tidy code - would be good to have an option to disable this as it can lead to wonky results if you're not keeping on top of adding them. Feel free to let me know if there is a way to do this already/why this isn't already a feature happy to learn that I'm wrong!

**Comment by oskarkrawczyk** [comment]
Yes, Prettier is... pretty opinionated in this matter. So for now it removes semicolons since it's a popular way of coding these days (including mine), but I'll add an option for the user to pick their way of coding.

**Comment by Dropboy** [comment]
Understood! Thanks for the update!

**Comment by mplungjan** [comment]
Please do not remove semicolons unless I am asking for it. I want to minify my code and cannot if I forget that jsfiddle removes most of them. I do not care that it is a popular way to code, it is not MY way of coding and I have on several occasions just wanted my code to be indented before I paste the code into an editor that does not indent or lint as well as jsfiddle used to.

**Comment by oskarkrawczyk** [comment]
@mplungjan Sure, I'll make the semicolons configurable at some point but don't get your hopes up for total configuration that adheres to your specific coding style - I won't be adding more configuration options than the semicolons.

**Comment by mplungjan** [comment]
I do not have any other needs in that regard. Thanks

**Comment by mplungjan** [comment]
Basically don't remove or add stuff other than indentation

**Comment by mplungjan** [comment]
Basically don't remove or add stuff other than indentation

**Comment by Dropboy** [comment]
Thanks for your work!

**Comment by mplungjan** [comment]
The "prettier" icon does not seem to do anything. Right-click and "format document" works as intended afaics

**Comment by mplungjan** [comment]
@oskarkrawczyk  How/where do I activate/deactivate the semi removal? Semis re currently removed and I do not see it in the editor preferences

**Comment by oskarkrawczyk** [comment]
@mplungjan https://github.com/jsfiddle/jsfiddle-users/issues/1896#issuecomment-2983593963

**Comment by mplungjan** [comment]
So it fails - how to convince it? Or at least have a setting to reverse the default which I personally hate

Code I want to format with two space indent

```
export function createNewItem() {
    const newItem = document.createElement('div');
    newItem.classList.add('new-div');

    const newHeading = document.createElement('h2');
    newHeading.classList.add('new-heading');

    const deleteButton = document.createElement('button');
    deleteButton.classList.add('delete-btn');
    deleteButton.innerText = 'Delete';

    const inputArea = document.createElement('input');
    inputArea.classList.add('textbox');
    inputArea.type = 'text';
    inputArea.placeholder = 'Enter here...';

    const main = document.querySelector('.main');
    const num = main.children.length;
    newHeading.innerText = `Task ${num}`;

    newItem.append(newHeading, deleteButton, inputArea);

    return newItem;
}
```
After prettifying - the indentaion is fixed but the quotes and semis are "fixed" - Please allow me to turn that off or change default

```
export function createNewItem() {
  const newItem = document.createElement("div")
  newItem.classList.add("new-div")

  const newHeading = document.createElement("h2")
  newHeading.classList.add("new-heading")

  const deleteButton = document.createElement("button")
  deleteButton.classList.add("delete-btn")
  deleteButton.innerText = "Delete"

  const inputArea = document.createElement("input")
  inputArea.classList.add("textbox")
  inputArea.type = "text"
  inputArea.placeholder = "Enter here..."

  const main = document.querySelector(".main")
  const num = main.children.length
  newHeading.innerText = `Task ${num}`

  newItem.append(newHeading, deleteButton, inputArea)

  return newItem
}
```

**Comment by oskarkrawczyk** [comment]
@mplungjan Please post this as a new ticket - I'll see why the check insists on using the defaults.
